draco/analysis/sidereal.py

```diff
@@ -477,6 +479,140 @@ def _regrid(self, vis, weight, lsd):
         return interp_grid, interp_vis, interp_weight
 
 
+class SiderealRebinner(SiderealRegridder):
+    """Regrid onto the sidereal day using a nearest bin method."""
+
+    def process(self, data):
+        """Rebin the sidereal day.
+
+        Parameters
+        ----------
+        data : containers.TimeStream
+            Timestream data for the day (must have a `LSD` attribute).
+
+        Returns
+        -------
+        sdata : containers.SiderealStream
+            The regularly gridded sidereal timestream.
+        """
+        import scipy.sparse as ss
+
+        self.log.info(f"Regridding LSD:{data.attrs['lsd']:.0f}")
+
+        # Redistribute if needed too
+        data.redistribute("freq")
+
+        # Convert data timestamps into LSDs
+        timestamp_lsd = self.observer.unix_to_lsd(data.time)
+
+        # Fetch which LSD this is to set bounds
+        self.start = data.attrs["lsd"]
+        self.end = self.start + 1
+
+        # Create the output container
+        sdata = containers.SiderealStreamRebin(axes_from=data, ra=self.samples)
+        sdata.redistribute("freq")
+        sdata.attrs["lsd"] = self.start
+        sdata.attrs["tag"] = f"lsd_{self.start:.0f}"
+
+        # Get view of data
+        weight = data.weight[:].local_array
+        vis_data = data.vis[:].local_array
+
+        # Get the average weights over baselines
+        average_weight = weight.mean(axis=1)
+        # Get the median time sample width
+        width_t = np.median(np.abs(np.diff(timestamp_lsd)))
+        # Create the regular grid of RA samples
+        target_lsd = np.linspace(self.start, self.end, self.samples, endpoint=False)
+
+        R = self._rebin_matrix(timestamp_lsd, target_lsd, width_t=width_t)
+        Rt = ss.csr_array(R.T)
+
+        # Calculate the total weight for each sample (which we track), and the
+        # normalisation we need to apply
+        total_weight = sdata.datasets["rebin_weight"][:].local_array
+        total_weight[:] = average_weight @ Rt
+        norm = tools.invert_no_zero(total_weight)
+
+        # Calculate the effective RA of each output sample given what went in
+        effective_lsd = norm * ((timestamp_lsd[np.newaxis, :] * average_weight) @ Rt)
+        sdata.datasets["effective_ra"][:].local_array[:] = 360 * (
+            effective_lsd - self.start
+        )
+
+        ssv = sdata.vis[:].local_array
+        ssw = sdata.weight[:].local_array
+
+        for fi in range(vis_data.shape[0]):
+            w = average_weight[fi, np.newaxis]
+            n = norm[fi, np.newaxis]
+            ssv[fi] = n * ((vis_data[fi] * w) @ Rt)
+
+            regrid_var = tools.invert_no_zero(weight[fi]) * (w @ Rt) ** 2
+            ssw[fi] = tools.invert_no_zero(n**2 * regrid_var)
+
+        return sdata
+
+    def _rebin_matrix(
```

Comment:
The `rebin_matrix` method could be moved into `util.regrid` since it does not require access to any class attributes.